### PR TITLE
Feature/conversions

### PR DESCRIPTION
**Pull request by fivetran-reneeli**
## PR Overview
**This PR will address the following Issue/Feature:**
Adding conversion metrics 

**This PR will result in the following new package version:**

v0.4.0

breaking change to users who weren't bringing in conversion metrics already through the passthrough variable.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Feature update: Conversion Support
- We have added conversion metrics to the following staging models:
  - ad_group_level_report
  - advertised_product_report
  - campaign_level_report
  - targeting_keyword_report
  - search_term_ad_keyword_report

- The conversion metrics are the following:
  - `purchases_30_d`: Number of attributed conversion events occurring within 30 days of an ad click.
  - `sales_30_d`: Total value of sales occurring within 30 days of an ad click.

## Under the hood: Backwards compatibility
- In the event that you were already passing the above fields in via our [passthrough columns](https://github.com/fivetran/dbt_google_ads_source?tab=readme-ov-file#passing-through-additional-metrics), the package will dynamically avoid "duplicate column" errors.
- This was done via the new `google_ads_fill_pass_through_columns` and `google_ads_add_pass_through_columns` macros to ensure that the new conversion fields are backwards compatible with users who have already included them via passthrough fields.

> The above new field additions are 🚨 **breaking changes** 🚨 for users who were not already bringing in conversion fields via passthrough columns.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [na] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [ ] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: integration_tests/requirements.txt [hunk 1]
```diff
@@ -4,4 +4,4 @@ dbt-redshift>=1.3.0,<2.0.0
 dbt-postgres>=1.3.0,<2.0.0
 dbt-spark>=1.3.0,<2.0.0
 dbt-spark[PyHive]>=1.3.0,<2.0.0
-dbt-databricks>=1.3.0,<2.0.0
\ No newline at end of file
+dbt-databricks>=1.6.0,<2.0.0
```
Comment (fivetran-joemarkiewicz) [inline integration_tests/requirements.txt:7]: Same question about if there is a reason this is different than the other versions

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM

File: integration_tests/requirements.txt (hunk 1 above)
Comment (fivetran-jamie) [reply integration_tests/requirements.txt:7]: Oh no real reason -- I just noticed this was inconsistent with some other packages and updated it accordingly. I can revert this if preferable!

Comment (fivetran-joemarkiewicz) [reply integration_tests/requirements.txt:7]: No this is okay to keep, I just wanted to make sure there wasn't a specific reason and if we needed to adjust our dbt required version for this update. All good to keep. Thanks!

**Pull request merged**